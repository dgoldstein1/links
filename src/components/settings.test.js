import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import Settings from "./settings";

const mockStore = configureStore([]);

describe("Settings", () => {
  const defaultState = {
    graph: {
      maxNeighbors: 15
    },
    appState: {
      language: "english"
    }
  };
  describe("matches snapshots", () => {
    let testTable = [
      {
        name: "success",
        state: defaultState
      }
    ];

    testTable.forEach(t => {
      it(t.name, () => {
        let store = mockStore(t.state);
        let wrapper = renderer.create(
          <Provider store={store}>
            <Settings />
          </Provider>
        );
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
  describe("calls onChange for language", () => {
    let store = mockStore(defaultState);
    let wrapper = renderer.create(
      <Provider store={store}>
        <Settings />
      </Provider>
    );
    // emulate click on button
    let t;
    renderer.act(() => {
      let event = {
        target: {
          value: "english"
        }
      };
      t = wrapper.root
        .findByProps({ id: "language-select" })
        .props.onChange(event);
    });
    expect(t).toMatchSnapshot();
  });
  describe("calls onChange for set max neighbors", () => {
    let store = mockStore(defaultState);
    let wrapper = renderer.create(
      <Provider store={store}>
        <Settings />
      </Provider>
    );
    // emulate click on button
    let t;
    renderer.act(() => {
      let event = {
        target: {
          value: 234234
        }
      };
      t = wrapper.root
        .findByProps({ id: "neighbor-limit" })
        .props.onChange(event);
    });
    expect(t).toMatchSnapshot();
  });
});
