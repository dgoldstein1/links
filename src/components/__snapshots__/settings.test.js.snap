// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "language": "english",
  "type": "SET_LANGUAGE",
}
`;

exports[` 2`] = `
Object {
  "maxNeighbors": 234234,
  "type": "SET_MAX_NEIGHBORS",
}
`;

exports[`Settings matches snapshots success 1`] = `
<div
  className="about-and-settings"
>
  <h1>
    Settings
  </h1>
  <form>
    <fieldset>
      <legend>
        General
      </legend>
      <label
        for="language-select"
      >
        Language
      </label>
      <select
        id="language-select"
        onChange={[Function]}
        value="english"
      >
        <option
          value="english"
        >
          English
        </option>
      </select>
    </fieldset>
    <fieldset>
      <legend>
        Graph
      </legend>
      <label
        for="neighbor-limit"
      >
        Maximum number of neighbors per node
      </label>
      <input
        id="neighbor-limit"
        max="250"
        min="1"
        onChange={[Function]}
        type="number"
        value={15}
      />
      <br />
      <label
        for="max-shortest-paths"
      >
        Maximum number of shortest paths
      </label>
      <input
        id="max-shortest-paths"
        max="250"
        min="1"
        onChange={[Function]}
        type="number"
      />
    </fieldset>
    <button>
      <a
        href="https://github.com/dgoldstein1/links/issues/new"
      >
        Report issue or bug
      </a>
    </button>
  </form>
  <footer
    className="App-footer"
  >
    Â© 2019 David Goldstein |
     
    <a
      href="http://davidcharlesgoldstein.com?ref=links-ui"
    >
      Personal Website
    </a>
     
    | 
    <a
      href="/LICENSE"
    >
      License
    </a>
     | 
    <a
      href="/VERSION"
    >
      Version
    </a>
  </footer>
</div>
`;
